collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java

```diff
@@ -15,295 +15,94 @@
+        return binding.widgetAnswerText.getText().equals(getContext().getString(R.string.no_time_selected))
+                ? null
+                : DateTimeWidgetUtils.getTimeData(selectedTime);
     }
 
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
-        timeButton.setOnLongClickListener(l);
-        timeTextView.setOnLongClickListener(l);
+        binding.widgetButton.setOnLongClickListener(l);
+        binding.widgetAnswerText.setOnLongClickListener(l);
     }
 
     @Override
     public void cancelLongPress() {
         super.cancelLongPress();
-        timeButton.cancelLongPress();
-        timeTextView.cancelLongPress();
-    }
-
-    private void createTimeButton() {
-        timeButton = createSimpleButton(getContext(), getFormEntryPrompt().isReadOnly(), getContext().getString(R.string.select_time), getAnswerFontSize(), this);
-    }
-
-    private void addViews() {
-        LinearLayout linearLayout = new LinearLayout(getContext());
-        linearLayout.setOrientation(LinearLayout.VERTICAL);
-        linearLayout.addView(timeButton);
-        linearLayout.addView(timeTextView);
-        addAnswerView(linearLayout, WidgetViewUtils.getStandardMargin(getContext()));
-    }
-
-    public void setTimeLabel() {
-        nullAnswer = false;
-        timeTextView.setText(getAnswer().getDisplayText());
-    }
-
-    private void createTimePickerDialog() {
-        timePickerDialog = new CustomTimePickerDialog(getContext(), this, 0, 0);
-        timePickerDialog.setCanceledOnTouchOutside(false);
-
-        if (getFormEntryPrompt().getAnswerValue() == null) {
-            clearAnswer();
-        } else {
-            Date date = (Date) getFormEntryPrompt().getAnswerValue().getValue();
-
-            DateTime dateTime = new DateTime(date);
-            updateTime(dateTime, true);
-        }
-    }
-
-    public int getHour() {
-        return hourOfDay;
-    }
-
-    public int getMinute() {
-        return minuteOfHour;
-    }
-
-    public boolean isNullAnswer() {
-        return nullAnswer;
-    }
-
-    public void setTimeToCurrent() {
-        updateTime(DateTime.now(), false);
-    }
-
-    public void updateTime(DateTime dateTime) {
-        updateTime(dateTime, true);
-    }
-
-    public void updateTime(DateTime dateTime, boolean shouldUpdateLabel) {
-        updateTime(dateTime.getHourOfDay(), dateTime.getMinuteOfHour(), shouldUpdateLabel);
-    }
-
-    public void updateTime(int hourOfDay, int minuteOfHour, boolean shouldUpdateLabel) {
-        this.hourOfDay = hourOfDay;
-        this.minuteOfHour = minuteOfHour;
-
-        timePickerDialog.updateTime(hourOfDay, minuteOfHour);
-
-        if (shouldUpdateLabel) {
-            setTimeLabel();
-        }
+        binding.widgetButton.cancelLongPress();
+        binding.widgetAnswerText.cancelLongPress();
     }
 
     @Override
-    public void onTimeSet(TimePicker timePicker, int hourOfDay, int minute) {
-        timePicker.clearFocus();
```

Comment:
Removing this you have broken https://github.com/getodk/collect/issues/3948 so it's a regression. In order to fix this you need to do the same now in FormEnryActivity where `onTimeSet` is used. Unfortunately we don't have any test for it. Could you write one? Or at leas please file an issue to do that in the future.

Reply:
Oh sorry @grzesiek2010. I will implement the `viewModel` and then I'll write tests for the same.

Reply:
Still not fixed. Have you tested it?

Reply:
Sorry for this. I fixed this in `731dc65`